commercesearch/infrastructure/productsearch/inmemoryrepository.go

```diff
@@ -310,33 +312,33 @@ func (s *marketPlaceCodeSet) intersection(set2 []string) {
 	s.currentSet = result
 }
 
-//addCategoryPath - merges in the given categoryToAdd to the  passed currentExisting
+// addCategoryPath merges in the given categoryToAdd to the  passed currentExisting
 // also adds new categories to the reverse index
 func (r *InMemoryProductRepository) addCategoryPath(currentTreeNode *categoryDomain.TreeData, treeNodeToAdd *categoryDomain.TreeData) *categoryDomain.TreeData {
-	if r.categorTreeIndex == nil {
-		r.categorTreeIndex = make(map[string]*categoryDomain.TreeData)
+	if r.categoryTreeIndex == nil {
+		r.categoryTreeIndex = make(map[string]*categoryDomain.TreeData)
 	}
-	//if its the first node then make as current node
+	// if its the first node then make as current node
 	if currentTreeNode == nil {
 		clone := *treeNodeToAdd
```

Comment:
is this really a clone? seems confusing

Reply:
as soon as you dereference a struct and assign it to a new variable you basically copy the struct, this only works for simple structs that don't hold pointers themselves.

Reply:
https://play.golang.org/p/HOkZgosR4_-